src/openrct2/network/NetworkUser.cpp

```diff
@@ -209,23 +185,24 @@ const NetworkUser* NetworkUserManager::GetUserByName(const std::string& name) co
 {
     for (const auto& kvp : _usersByHash)
     {
-        const NetworkUser* networkUser = kvp.second;
+        const auto& networkUser = kvp.second;
         if (String::Equals(name.c_str(), networkUser->Name.c_str(), true))
         {
-            return networkUser;
+            return networkUser.get();
         }
     }
     return nullptr;
 }
 
 NetworkUser* NetworkUserManager::GetOrAddUser(const std::string& hash)
 {
-    NetworkUser* networkUser = GetUserByHash(hash);
+    NetworkUser* networkUser = const_cast<NetworkUser*>(GetUserByHash(hash));
```

Comment:
If something is marked const you probably don't want to modify it. Declare it also const

Reply:
This change is to remove code duplication for `GetUserByHash` by keeping only const version impl.
Since `NetworkUserManager::GetOrAddUser` is mutable I had to const_cast it.

Reply:
networkUser is not modified, you modify newNetworkUser and then assign the pointer, so you can declare it const just fine.

Reply:
The caller of `NetworkUserManager::GetOrAddUser` is modifying returned network user.

https://github.com/OpenRCT2/OpenRCT2/blob/86cab9861c95decb06c4c0532f4baa348e4e1408/src/openrct2/network/NetworkBase.cpp#L389

Reply:
Or should I const_cast from the caller?

Reply:
It shouldn't have to do that, see https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es50-dont-cast-away-const
In this case its not needed
```cpp
const auto* networkUser = GetUserByHash(hash);
```
Should do the trick, the in case it returns null you create a new variable which is not const therefor you can modify the state and then later assign networkUser to the new pointer.

Reply:
If `GetUserByHash` returns nullptr, then yes we could indeed make a new non-const user and return it.
However, if it does not return nullptr, then we have to return `const NetworkUser*`, which makes it impossible for the caller to modify the state.
I think it is because `GetOrAddUser` actually can either return a new pointer or existing pointer.

Reply:
Its a bit strange that  GetUserByHash returns a const version and GetOrAddUser does not. I guess leave it be, this needs to be looked at in another PR I think.